pkg/testing/ig/ig.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ig provides executable wrapper for ig binary.
+//
+// Mainly used for testing of image-based gadgets.
+package ig
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	igtesting "github.com/inspektor-gadget/inspektor-gadget/pkg/testing"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/command"
+)
+
+// runner is responsible for storing configuration of ig executable and provide methods to interact with.
+type runner struct {
+	path  string
+	image string
+
+	// command.Command contains *exec.Cmd and additional properties and methods for the same.
+	command.Command
+	flags []string
+}
+
+func (ig *runner) createCmd() {
+	ig.flags = append(ig.flags, "-o=json")
+	args := append([]string{"run", ig.image}, ig.flags...)
+
+	ig.Cmd = exec.Command(ig.path, args...)
+}
+
+type option func(*runner)
+
+// WithPath used for providing custom path to ig executable.
+func WithPath(path string) option {
+	return func(ig *runner) {
+		ig.path = path
+	}
+}
+
+// WithPathFromEnvVar used for reading custom path for ig executable from IG env var.
+func WithPathFromEnvVar() option {
+	return func(ig *runner) {
+		ig.path = os.Getenv("IG")
+	}
+}
```

Comment:
IMO we don't need this. We can hardcode it to always use the value on the `IG` if it's defined. By doing so we:
- Remove the needed use `WithPathFromEnvVar` in the test
- Remove the needed to set `IG` when running the test

I'm suggesting something like: (It's possible this diff doesn't apply because I edited it manually, but that's the idea)

```diff
diff --git a/gadgets/trace_open/test/trace_open_test.go b/gadgets/trace_open/test/trace_open_test.go
index 4a2367028..8ae964871 100644
--- a/gadgets/trace_open/test/trace_open_test.go
+++ b/gadgets/trace_open/test/trace_open_test.go
@@ -19,7 +19,6 @@ import (

        "github.com/stretchr/testify/require"

-       gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
        igtesting "github.com/inspektor-gadget/inspektor-gadget/pkg/testing"
        "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/containers"
        igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
@@ -44,15 +43,12 @@ type traceOpenEvent struct {
 }

 func TestTraceOpen(t *testing.T) {
-       gadgettesting.RequireEnvironmentVariables(t)
-
        cn := "test-trace-open"
        containerFactory, err := containers.NewContainerFactory("docker")
        require.NoError(t, err, "new container factory")

        traceOpenCmd := igrunner.New(
                "trace_open",
-               igrunner.WithPathFromEnvVar(),
                igrunner.WithFlags("--runtimes=docker", "--timeout=10"),
                igrunner.WithValidateOutput(
                        func(t *testing.T, output string) {
diff --git a/pkg/testing/ig/ig.go b/pkg/testing/ig/ig.go
index 5bf23a987..45aa0399b 100644
--- a/pkg/testing/ig/ig.go
+++ b/pkg/testing/ig/ig.go
@@ -53,13 +53,6 @@ func WithPath(path string) option {
        }
 }

-// WithPathFromEnvVar used for reading custom path for ig executable from IG env var.
-func WithPathFromEnvVar() option {
-       return func(ig *runner) {
-               ig.path = os.Getenv("IG")
-       }
-}
-
 // WithFlags args should be in form: "--flag_name=value" or "-shorthand=value".
 func WithFlags(flags ...string) option {
        return func(ig *runner) {
@@ -83,14 +76,24 @@ func WithValidateOutput(validateOutput func(t *testing.T, output string)) option

 // New creates a new IG configured with the options passed as parameters.
 func New(image string, opts ...option) igtesting.TestStep {
        ig := &runner{
                path:  "ig",
                Command: command.Command{
                        Name: "Run_" + image,
                },
        }

+       if path, ok := os.LookupEnv("IG"); ok {
+               ig.path = path
+       }
+
        for _, opt := range opts {
                opt(ig)
        }
```

Reply:
Right, we can remove the first part, but the second part would still be required since we want to skip this test when we use `make test`, instead of using a separate flag we check whether IG is defined to determine whether we skip the test.

Reply:
Makes sense to me. Let's skip my comment for now then.